src/Ouroboros-Mini_Protocols-Chain_Sync.thy

```diff
@@ -0,0 +1,178 @@
+section \<open>Chain Synchronization Mini-Protocol\<close>
+
+text \<open>
+  The theory in this section implements the chain synchronization mini-protocol.
+\<close>
+
+theory "Ouroboros-Mini_Protocols-Chain_Sync"
+  imports
+    "Ouroboros-Mini_Protocols"
+    "HOL-Library.BNF_Corec"
+begin
+
+locale chain_sync =
+  fixes initial_items\<^sub>c :: "'i list"
+  fixes initial_items\<^sub>s :: "'i list"
+  fixes candidate_points :: "'i list \<Rightarrow> 'p list"
```

Comment:
Good question. I don't like using non-single-letter identifiers for type parameters (e.g., `pt` or `point`) unless it's strictly necessary. Then, how about using `π` instead?

Reply:
I’m a bit reluctant regarding Greek letters in type variables. What about `q`? It’s just a `p` mirrored. :wink: In any case, when changing type parameters `'p` to something else, we should also change term variables `p` accordingly.

Reply:
I kind of like Greek letters in type variables, perhaps because that's the style used in _Introduction to Functional Programming_ by Bird and Wadler, which was the book with which I was introduced to functional programming back in college. :heart_eyes:

So, would you suggest to use `q` instead of `p`?

Reply:
> I kind of like Greek letters in type variables, perhaps because that's the style used in _Introduction to Functional Programming_ by Bird and Wadler, which was the book with which I was introduced to functional programming back in college. 😍

Well, in books and papers, using α, β, etc. is common, but in real Haskell code you typically have `a`, `b`, etc.
 
> So, would you suggest to use `q` instead of `p`?

Yes, let’s go for `q`.

Reply:
> Yes, let’s go for `q`.

Fixed in 7a74e4dedd4729e84d2b4c58667a1ffb9819a151, please check.